### PR TITLE
DM-8561: Filter None-valued measurements

### PR DESCRIPTION
**Pull request by jonathansick**
- Don't print JSON after SQUASH POST failure
- Add --metrics CLI flag: allows us to customize what metrics are submitted from the command line invocation. Also removes AM3 from the default upload list.
- Omit measurements with None values from upload: needed for SQUASH API compatibility.
- Centralize dependencies in setup.py.

### CODE REVIEW

File: postqa/cli.py [hunk 1]
```diff
@@ -105,7 +105,6 @@ def build_job_json(qa_json_path, lsstsw_dirname):
 def upload_json(job_json, api_url, api_user, api_password):
     """Upload Job json document to SQuaSH through POST /api/jobs/ endpoint."""
     r = requests.post(api_url, auth=(api_user, api_password), json=job_json)
-    print(r.status_code)
+    print('POST {0} status: {1}'.format(api_url, r.status_code))
```
Comment (afausti) [inline postqa/cli.py]: @jonathansick I've tested some conditions like
1) wrong URL
POST http://localhost:8000/dashboard/api/jobs status: 500

2) wrong user/password 
POST http://localhost:8000/dashboard/api/jobs/ status: 401

Perhaps we could catch  exceptions like ConnectionError or Timeout, but I think this is fine.

File: postqa/jsonshim.py [hunk 2]
```diff
@@ -15,18 +15,15 @@ def shim_validate_drp(vdrp_json, accepted_metrics):
...
-       either as input or output.
     """
     job_json = {}
 
     measurements = []
     for vdrp_measurement_doc in vdrp_json['measurements']:
         if vdrp_measurement_doc['metric']['name'] not in accepted_metrics:
             continue
+        if vdrp_measurement_doc['value'] is None:
+            continue
```
Comment (afausti) [inline postqa/jsonshim.py:26]: I think it make sense to skip metrics with None value. For a given metric if we have consecutive jobs with missing measurements this is highlighted in the time series plot and might indicate that 1) validade_drp run failed or 2) the dataset is not suitable to compute a specific metric.  I think we are adding 2) now.

**Comment by afausti** [comment]
@jonathansick tested this ticket branch, including the new --metric flag and it is working fine.

File: postqa/cli.py (hunk 1 above)
Comment (jonathansick) [reply postqa/cli.py]: Good idea. I've added stuff to `upload_json` to force requests to raise an exception, and then print out that exception. That'll catch ConnectionErrors and Timouts.

**Pull request merged**